wp-content/themes/pub/wporg-learn-2024/src/course-outline/index.js

```diff
@@ -0,0 +1,41 @@
+/* global wporgCourseOutlineData */
+
+import { Icon, drafts, lockOutline } from '@wordpress/icons';
+import { renderToString } from 'react-dom/server';
+
+document.addEventListener( 'DOMContentLoaded', function () {
+	const lessonData = wporgCourseOutlineData;
+
+	lessonData[ 'in-progress' ]?.forEach( function ( lessonTitle ) {
+		const title = lessonTitle;
+		const icon = renderToString( <Icon icon={ drafts } transform={ 'scale(1.5)' } /> );
+
+		const lessonLinks = document.querySelectorAll( '.wp-block-sensei-lms-course-outline-lesson' );
+
+		lessonLinks.forEach( function ( link ) {
+			const span = link.querySelector( 'span' );
+
+			if ( span && span.textContent.trim() === title ) {
+				const statusIcon = link.querySelector( '.wp-block-sensei-lms-course-outline-lesson__status' );
+				if ( statusIcon ) {
+					statusIcon.outerHTML = icon;
+				}
+			}
+		} );
+	} );
+
+	lessonData.locked?.forEach( function ( lessonTitle ) {
+		const title = lessonTitle;
+		const icon = renderToString( <Icon icon={ lockOutline } /> );
+
+		const lessonLinks = document.querySelectorAll( '.wp-block-sensei-lms-course-outline-lesson' );
+
+		lessonLinks.forEach( function ( link ) {
+			const span = link.querySelector( 'span' );
+
+			if ( span && span.textContent.trim() === title ) {
+				span.insertAdjacentHTML( 'afterend', icon );
+			}
+		} );
+	} );
+} );
```

Comment:
Nitpicky, but I think this could be made more concise, by removing unnecessary consts and using arrow functions:
```suggestion
document.addEventListener( 'DOMContentLoaded', () => {
	wporgCourseOutlineData[ 'in-progress' ]?.forEach( ( title ) => {
		const lessonLinks = document.querySelectorAll( '.wp-block-sensei-lms-course-outline-lesson' );

		lessonLinks.forEach( ( link ) => {
			const span = link.querySelector( 'span' );

			if ( span && span.textContent.trim() === title ) {
				const statusIcon = link.querySelector( '.wp-block-sensei-lms-course-outline-lesson__status' );

				if ( statusIcon ) {
					statusIcon.outerHTML = renderToString( <Icon icon={ drafts } transform={ 'scale(1.5)' } /> );
				}
			}
		} );
	} );

	wporgCourseOutlineData.locked?.forEach( ( title ) => {
		const lessonLinks = document.querySelectorAll( '.wp-block-sensei-lms-course-outline-lesson' );

		lessonLinks.forEach( function ( link ) {
			const span = link.querySelector( 'span' );

			if ( span && span.textContent.trim() === title ) {
				span.insertAdjacentHTML( 'afterend', renderToString( <Icon icon={ lockOutline } /> ) );
			}
		} );
	} );
} );
```

Reply:
Not at all, it's nice that you brought this up. I appreciate that we have thoughts and take this seriously. Will pay more attention to this moving forward.

fa10402